app/models/double-map.ts

```diff
@@ -0,0 +1,49 @@
+
+
+/**
+ * Keep two dictionaries in sync, different efficient maps to the same data
+ */
+export class DoubleMap<A, B, C> {
```

Comment:
If this is only used one place, then make it non-generic since generic code is inherently more complicated.